docs/tutorials/qsimcirq_gcp.md

```diff
@@ -0,0 +1,357 @@
+# Quantum simulation on GCP with _Cirq_ and _qsim_
+
+This tutorial will demonstrate how to run [Cirq](https://cirq.readthedocs.io/en/latest/index.html) on Google Cloud Platform. This tutorial will walk through how to install Cirq within a Docker container on a GCE Virtual Machine and view results. You will run simulation both in a Jupyter environment and interactively within the container.
+
+Explaining the concepts of quantum computing is beyond the scope of this tutorial, but many excellent references and texts exist. This website describes [Cirq](https://Cirq.readthedocs.io/en/stable/index.html) in detail. Additionally, the textbook *Quantum Computation and Quantum Information* by Nielsen and Chuang is an excellent reference.
+
+## Objectives
+
+*   Create a Container Optimized VM
+*   Run Docker container with Jupyter and qsim installed
+*   Run a demo circuit
+
+## Costs
+
+This tutorial uses billable components of Google Cloud Platform, including the *Compute Engine*. Use the [Pricing Calculator](https://cloud.google.com/products/calculator) to generate a cost estimate based on your projected usage.
+
+## Before you begin
+
+These are the steps to follow to get started with this tutorial:
+
+
+
+*   Creating a Cloud Platform project
+*   Enabling billing for the project
+*   Enable Google Compute Engine API
+
+For this reference guide, you need a Google Cloud [project](https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy#projects). You can create a new one, or select a project you already created:
+
+
+
+1. Select or create a Google Cloud project.
+
+> [GO TO THE PROJECT SELECTOR PAGE](https://console.cloud.google.com/projectselector2/home/dashboard)
+
+
+
+2. Enable billing for your project.
+
+> [ENABLE BILLING](https://support.google.com/cloud/answer/6293499#enable-billing)
+
+When you finish this tutorial, you can avoid continued billing by deleting the resources you created. See [Cleaning up](#cleaning-up) for more detail.
+
+
+## Creating a GCE VM 
+
+Once you have a project enabled with a billing account, you will be able to create resources. The key resource required for this project is a Virtual Machine (VM) used to run the qsim quantum simulations. 
+
+
+### Use the Cloud Console
+
+Connect to the Cloud Console and make sure your current project is selected. Then click on "Create" for a new VM instance:
+
+
+
+![alt_text](../images/image7.png )
+
+
+
+### Build a Container Optimized VM
+
+To create the VM use the steps in sequence below:
+
+
+*   Change the name of the VM to be something meaningful like "qsim-1".
+*   Choose a Region and Zone
+    *   The values are not too important now, but for latency and availability [you may make different choice](https://cloud.google.com/compute/docs/regions-zones#available).
+*   Choose the machine family / series: N2
```

Comment:
Family N2 wasn't an option for me. I saw E2 and N1 (chose N1 to continue to tutorial). n1-standard-16 completed the q32d14 circuit in ~160 seconds.

Reply:
Which zone are you in?

Reply:
Test was in `us-west2-a`, which claims to have N2 machines available. Maybe I lack certain permissions?

Either way, given that performance on N1 was comparable, we could list that as a viable alternative.

Reply:
It's possible there was a stockout, but I think that usually gets flagged by rejecting your VM create request, not by hiding the machine type.  I think most of the machine types will work fine, if they have enough memory. But for performance C2 > N2 > N1, etc.  I'll add a note.

Reply:
Comment from @jrossthomson:
> I didn't do anything other than create the VM. So I am wondering if
something else is going on. Is this a new VM created using the Container
Optimized VM?

Reply:
Yes, it was a new VM in a fresh GCP project using the container-optimized OS image. I may have set the image **after** choosing the machine type, if that makes a difference.